migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.lightblue.migrator;
+
+import java.io.IOException;
+
+import java.util.Map;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.lightblue.client.LightblueClient;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.hystrix.LightblueHystrixClient;
+import com.redhat.lightblue.client.expression.query.ValueQuery;
+
+import static com.redhat.lightblue.client.expression.query.ValueQuery.withValue;
+import static com.redhat.lightblue.client.projection.FieldProjection.includeFieldRecursively;
+
+public class Controller extends Thread {
+
+    private static final Logger LOGGER=LoggerFactory.getLogger(Controller.class);
+    
+    private final MainConfiguration cfg;
+    private final LightblueClient lightblueClient;
+    private final Map<String,MigrationProcess> migrationMap=new HashMap<>();
+
+
+    public static class MigrationProcess {
+        public final MigrationConfiguration cfg;
+        public final MigratorController mig;
+
+        public MigrationProcess(MigrationConfiguration cfg,
+                                MigratorController mig) {
+            this.cfg=cfg;
+            this.mig=mig;
+        }
+    }
+    
+    public Controller(MainConfiguration cfg) {
+        this.cfg=cfg;
+        this.lightblueClient=getLightblueClient();
+    }
+
+    public Map<String,MigrationProcess> getMigrationProcesses() {
+        return migrationMap;
+    }
+
+    public MainConfiguration getMainConfiguration() {
+        return cfg;
+    }
+    
+    /**
+     * Read configurations from the database whose name matches this instance name
+     */
+    public MigrationConfiguration[] getMigrationConfigurations()
+        throws IOException {
+        DataFindRequest findRequest = new DataFindRequest("migrationConfiguration",null);
+        findRequest.where(withValue("consistencyCheckerName = " + cfg.getName()));
+        findRequest.select(includeFieldRecursively("*"));
+        LOGGER.debug("Loading configuration:{}",findRequest.getBody());
+        return lightblueClient.data(findRequest, MigrationConfiguration[].class);
+    }
+    
+    public LightblueClient getLightblueClient() {
+        LightblueHttpClient httpClient;
+        if (cfg.getClientConfig() != null) {
+            httpClient = new LightblueHttpClient(cfg.getClientConfig());
+        } else {
+            httpClient = new LightblueHttpClient();
+        }
+        return new LightblueHystrixClient(httpClient, "migrator", "primaryClient");
+    }
```

Comment:
Any reason not to lazy load and just return the class instance if not null? The method name getLightblueClient() makes me think I would get the same client instance each time.